google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -372,13 +373,46 @@ void processReceivedMessages(List<ReceivedMessage> messages) {
         // totally expire so that pubsub service sends us the message again.
         continue;
       }
-      outstandingBatch.add(new OutstandingMessage(message, ackHandler));
-      pendingReceipts.add(ackRequestData);
+      if (this.exactlyOnceDeliveryEnabled.get()) {
+        exactlyOncePendingReceipts.add(ackRequestData);
+      } else {
+        outstandingBatch.add(new OutstandingMessage(message, ackHandler));
+        pendingReceipts.add(ackRequestData);
+      }
+    }
+    if (this.exactlyOnceDeliveryEnabled.get()) {
+      List<ModackRequestData> modackRequestData = new ArrayList<ModackRequestData>();
+      List<AckRequestData> ackRequestDataReceipts = new ArrayList<AckRequestData>();
+      exactlyOncePendingReceipts.drainTo(ackRequestDataReceipts);
+      if (!ackRequestDataReceipts.isEmpty()) {
+        modackRequestData.add(
+            new ModackRequestData(this.getMessageDeadlineSeconds(), ackRequestDataReceipts));
+      }
+      ackProcessor.sendModackOperations(modackRequestData);
+      ApiFuture.addCallback(ackReqData.messageFuture, new ApiFutureCallback<string>() {
```

Comment:
I don't see `ackReqData` defined anywhere. The class no longer builds: https://github.com/googleapis/java-pubsub/actions/runs/4597320560/jobs/8119863303?pr=1540#step:5:623